src/Workspaces/CSharpTest/EmbeddedLanguages/RegularExpressions/CSharpRegexParserTests_BasicTests.cs

```diff
@@ -0,0 +1,15106 @@
+      <Text>
+        <TextToken>0}</TextToken>
+      </Text>
+    </Sequence>
+    <EndOfFile />
+  </CompilationUnit>
+  <Diagnostics>
+    <Diagnostic Message=""Quantifier {x,y} following nothing"" Start=""9"" Length=""1"" />
+  </Diagnostics>
+  <Captures>
+    <Capture Name=""0"" Span=""[9..12)"" />
+  </Captures>
+</Tree>", RegexOptions.None);
+        }
+
+        [Fact]
+        public void TestNumericQuantifierAtStartOfGroup()
+        {
+            Test(@"""({0})""", @"<Tree>
+  <CompilationUnit>
+    <Sequence>
+      <SimpleGrouping>
+        <OpenParenToken>(</OpenParenToken>
+        <Sequence>
+          <Text>
+            <TextToken>{</TextToken>
+          </Text>
+          <Text>
+            <TextToken>0}</TextToken>
+          </Text>
+        </Sequence>
+        <CloseParenToken>)</CloseParenToken>
+      </SimpleGrouping>
+    </Sequence>
+    <EndOfFile />
+  </CompilationUnit>
+  <Diagnostics>
+    <Diagnostic Message=""Quantifier {x,y} following nothing"" Start=""10"" Length=""1"" />
+  </Diagnostics>
+  <Captures>
+    <Capture Name=""0"" Span=""[9..14)"" />
+    <Capture Name=""1"" Span=""[9..14)"" />
+  </Captures>
+</Tree>", RegexOptions.None);
+        }
+
+        [Fact]
+        public void TestNumericQuantifierAfterQuantifier()
+        {
+            Test(@"""a*{0}""", @"<Tree>
+  <CompilationUnit>
+    <Sequence>
+      <ZeroOrMoreQuantifier>
+        <Text>
+          <TextToken>a</TextToken>
+        </Text>
+        <AsteriskToken>*</AsteriskToken>
+      </ZeroOrMoreQuantifier>
+      <Text>
+        <TextToken>{</TextToken>
+      </Text>
+      <Text>
+        <TextToken>0}</TextToken>
+      </Text>
+    </Sequence>
+    <EndOfFile />
+  </CompilationUnit>
+  <Diagnostics>
+    <Diagnostic Message=""Nested quantifier {"" Start=""11"" Length=""1"" />
+  </Diagnostics>
+  <Captures>
+    <Capture Name=""0"" Span=""[9..14)"" />
+  </Captures>
+</Tree>", RegexOptions.None);
+        }
+
+        [Fact]
+        public void TestNonNumericQuantifierAtStart()
+        {
+            Test(@"""{0""", @"<Tree>
+  <CompilationUnit>
+    <Sequence>
+      <Text>
+        <TextToken>{0</TextToken>
+      </Text>
+    </Sequence>
+    <EndOfFile />
+  </CompilationUnit>
+  <Captures>
+    <Capture Name=""0"" Span=""[9..11)"" />
+  </Captures>
+</Tree>", RegexOptions.None);
+        }
+
+        [Fact]
+        public void TestNonNumericQuantifierAtStartOfGroup()
+        {
+            Test(@"""({0)""", @"<Tree>
+  <CompilationUnit>
+    <Sequence>
+      <SimpleGrouping>
+        <OpenParenToken>(</OpenParenToken>
+        <Sequence>
+          <Text>
+            <TextToken>{0</TextToken>
+          </Text>
+        </Sequence>
+        <CloseParenToken>)</CloseParenToken>
+      </SimpleGrouping>
+    </Sequence>
+    <EndOfFile />
+  </CompilationUnit>
+  <Captures>
+    <Capture Name=""0"" Span=""[9..13)"" />
+    <Capture Name=""1"" Span=""[9..13)"" />
+  </Captures>
+</Tree>", RegexOptions.Non
```

Comment:
I'm going to pause here. Reviewed a decent sample from this file. I'll take a look at the other files tomorrow.